compiler/src/main/kotlin/walkers/type.kt

```diff
@@ -0,0 +1,333 @@
+package dk.aau.cs.d409f19.cellumata.walkers
+
+import dk.aau.cs.d409f19.cellumata.ast.*
+import kotlin.AssertionError
+
+/**
+ * Thrown when a violation of the type rules is found
+ */
+class TypeError : Exception()
+
+/**
+ * Move types up the abstract syntax tree according to the type rules, and check that there is no violation of the type rules
```

Comment:
Rather "synthesizes", as this is what our SPO-course teaches as terminology?

Reply:
> Synthesizes types by moving them up the abstract syntax tree according to the type rules, and check that there is no violation of the type rules

How about this?